### PR TITLE
Custom Notification Dot Counter Color Support

### LINKED ISSUE #2923: Option to have a white notification count
### Describe the feature

Would it be possible to add the option to have the number in the notification count written in white instead of black?

### Why would it be useful to add?

It would make the readability better for dark colored dot notifications.

### Did the feature exist in Lawnchair v2? (Play Store version)

No

### Example(s)

Screenshot_20220909-184315
Screenshot_20220909-181822

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by yasanglass**
## Description

Adds custom color support to Notification Dots' counter text.

Requires [platform_frameworks_libs_systemui#6](https://github.com/LawnchairLauncher/platform_frameworks_libs_systemui/pull/6).

Fixes #2923

## Type of change

:x: General change (non-breaking change that doesn't fit the below categories like copyediting)
:x: Bug fix (non-breaking change which fixes an issue)
:white_check_mark: New feature (non-breaking change which adds functionality)
:x: Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Comment by yasanglass** [comment]
5ac959a71e50a6feff03a25438c6888791f30f02 Adds a warning that is shown whenever the dot background & foregound (counter text) colors do not have enough contrast with each other.

[ColorContrastWarning](https://github.com/LawnchairLauncher/lawnchair/pull/2977/commits/5ac959a71e50a6feff03a25438c6888791f30f02#diff-896a87a728219ae3a62e4b9efa7f2b25894b68445ec66b5d45e0fe68402bddfd) composable can be used for any other two colors when needed later on.

Screenshot_20220926_122902

**Comment by yasanglass** [comment]
647f8efcbfed4e7f3aa32ba7799c9705cb535dde changes `ColorDot` so it shows an "A" icon when the color is _**Managed by Lawnchair**_ (color is 0). 

Screenshot from 2022-09-26 17-15-53Screenshot from 2022-09-26 17-15-44

**Comment by yasanglass** [comment]
d84b8ea441943eba477ae03f50c485048ae4a440 adds a new custom scenario for when the contrast warning is shown:

The `default` color for dot & dot text colors behave differently:
   - default dot color -> picked automatically based on icon color
   - default dot text color -> picked automatically to have enough contrast with the dot color

As a result, if the dot color is dynamic (changes based on the app icon, system theme, or wallpaper) but the text color is not `default` (picked automatically based on the dot color), the colors can sometimes **not** have enough contrast with each other in some scenarios.

Screenshot from 2022-09-26 17-43-38

File: lawnchair/src/app/lawnchair/ui/preferences/components/colorpreference/ColorContrast Warning.kt [hunk 1]
```diff
@@ -0,0 +1,63 @@
+package app.lawnchair.ui.preferences.components.colorpreference
```
Comment (suphon-t) [inline lawnchair/src/app/lawnchair/ui/preferences/components/colorpreference/ColorContrast Warning.kt]: Looks like there's a space in the file name

File: lawnchair/src/app/lawnchair/ui/preferences/components/WarningPreference.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        title = {},
+        description = {
+            Text(
+                text = text,
+                color = MaterialTheme.colorScheme.error,
+            )
+        },
+        startWidget = {
+            Icon(
+                imageVector = Icons.Outlined.Warning,
```
Comment (suphon-t) [inline lawnchair/src/app/lawnchair/ui/preferences/components/WarningPreference.kt]: Please change it to use the rounded icons to be consistent with other part of the UI.

Comment (yasanglass) [reply lawnchair/src/app/lawnchair/ui/preferences/components/WarningPreference.kt]: My bad, fixed.

File: lawnchair/src/app/lawnchair/ui/preferences/components/colorpreference/ColorContrast Warning.kt (hunk 1 above)
Comment (yasanglass) [reply lawnchair/src/app/lawnchair/ui/preferences/components/colorpreference/ColorContrast Warning.kt]: That's weird. Fixed it now.

File: lawnchair/src/app/lawnchair/ui/preferences/components/colorpreference/ColorDot.kt [hunk 3]
```diff
@@ -17,27 +21,53 @@ fun <T> ColorDot(
     entry: ColorPreferenceEntry<T>,
     modifier: Modifier = Modifier
 ) {
+    val context = LocalContext.current
+
+    val colorLight = entry.lightColor(context)
+    val colorDark = entry.darkColor(context)
```
Comment (Goooler) [inline lawnchair/src/app/lawnchair/ui/preferences/components/colorpreference/ColorDot.kt:27]: Unnecessary change?

Comment (yasanglass) [reply lawnchair/src/app/lawnchair/ui/preferences/components/colorpreference/ColorDot.kt:27]: I created the variables because I needed to use `colorLight` on line 35.

**Review by suphon-t — APPROVED** [review]

**Review by Goooler — APPROVED** [review]
